flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/SqlGatewayServiceImpl.java

```diff
@@ -79,6 +82,32 @@ public void closeSession(SessionHandle sessionHandle) throws SqlGatewayException
         }
     }
 
+    @Override
+    public ResultSet configureSession(
+            SessionHandle sessionHandle, String statement, long executionTimeoutMs)
+            throws SqlGatewayException {
+        try {
+            if (executionTimeoutMs > 0) {
+                // TODO: support the feature in FLINK-27838
+                throw new UnsupportedOperationException(
+                        "SqlGatewayService doesn't support timeout mechanism now.");
+            }
+            OperationHandle operationHandle =
+                    getSession(sessionHandle)
+                            .getOperationManager()
+                            .submitOperation(
+                                    handle ->
+                                            getSession(sessionHandle)
+                                                    .createExecutor()
+                                                    .configureSession(handle, statement));
+            return fetchConfigureSessionResult(sessionHandle, operationHandle);
```

Comment:
Are there any concurrency concerns here?

I'm new to the codebase; seems like it ought to be ok, but figured I'd ask.

To validate what I understand so far, I'm guessing that the SQL client will block on getting a response back.  Also, it looks like the `fetchConfigureSessionResult` will block on getting the result back.

Reply:
For the Operation (not the flink table operation; it is the gateway service operation) submitted to `OperationManager`, it has to get lock (see [operationLock](https://github.com/apache/flink/blob/3c9a394c608125827d224180c44128718972a82c/flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/operation/OperationManager.java#L66) before running. And we make sure that the configure session returns after the execution result is ready, so concurrency problem won't happen here (for example, if two clients holds the same `SessionHandle` somehow and submit two conflict configuring statement, one should succeed and then another should fail).